### PR TITLE
docs: add scheduler documentation

### PR DESCRIPTION
**Pull request by zhiying-lin**
### Description of your changes

Fixes #

I have:

- [ ] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: docs/concepts/Scheduler/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    type: Applied
+```
+
+`ClusterResourceBinding` can have three states:
+* _Scheduled_: It indicates that the scheduler has selected this cluster for placing the resources. The resource is waiting
+to be picked up by the rollout controller.  
+* _Bound_: It indicates that the rollout controller has initiated the placement of resources on the target cluster. The
+resources are actively being deployed.
+* _Unscheduled_: This states signifies that the target cluster is no longer selected by the scheduler for the placement.
+The resource associated with this cluster are in the process of beinng removed. They are awaiting deletion from the cluster.
```
Comment (britaniar) [inline docs/concepts/Scheduler/README.md:82]: There is an extra n in being, should only be one

File: docs/concepts/Scheduler/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+1. In a `ClusterResourcePlacement`, multiple replicas of resources cannot be scheduled on the same cluster, whereas pods
+belonging to the same deployment in Kubernetes can run on the same node.
+2. The `ClusterResourcePlacement` supports different placement types within a single object.
+
+These requirements necessitate treating the scheduling policy as a while and feeding it to the scheduler, as opposed to 
+handling individual pods like Kubernetes today. Specially:
+1. Scheduling the entire `ClusterResourcePlacement` at once enables us to increase the parallelism of the scheduler if
+needed.
+2. Supporting the `PickAll` mode would require generating the replica for each cluster in the fleet to scheduler. This
```
Comment (britaniar) [inline docs/concepts/Scheduler/README.md:21]: If it is inefficient, what is the alternative approach for this? Is this a lead up to the statement about supporting PickN?

File: docs/concepts/Scheduler/README.md (hunk 1 above)
Comment (ryanzhang-oss) [inline docs/concepts/Scheduler/README.md:82]:
```suggestion
The resource associated with this cluster are in the process of being removed. They are awaiting deletion from the cluster.

```

File: docs/concepts/Scheduler/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+## Batch in nature
+
+Scheduling resources within a `ClusterResourcePlacement` involves more dependencies compared with scheduling pods within
+a deployment in Kubernetes. There are two notable distinctions:
+
+1. In a `ClusterResourcePlacement`, multiple replicas of resources cannot be scheduled on the same cluster, whereas pods
+belonging to the same deployment in Kubernetes can run on the same node.
+2. The `ClusterResourcePlacement` supports different placement types within a single object.
+
+These requirements necessitate treating the scheduling policy as a while and feeding it to the scheduler, as opposed to
```
Comment (ryanzhang-oss) [inline docs/concepts/Scheduler/README.md:17]:
```suggestion
These requirements necessitate treating the scheduling policy as a whole and feeding it to the scheduler, as opposed to 

```

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**